File: src/main/scala/simple/LongestCommonPrefix.scala
package simple


object LongestCommonPrefix extends App{

  val st = Array("abcxy","abchh","abcyhj")
  longestCommonPrefix(st)

  def longestCommonPrefix(strs: Array[String]): String = {
  if (strs==null || strs.length==0){
    return ""
  }
    var pre = strs(0)
    var i =0
    while(i<strs.length){
     val s= (strs(i).indexOf(pre))
      while(strs(i).indexOf(pre) != 0)
        pre = pre.substring(0,pre.length()-1)
      i=i+1

    }
    pre
  }

}

File: src/main/java/ad/LongestCommonPrefix.java
package ad;

public class LongestCommonPrefix {
    public static void main(String[] args) {
        String [] input = {"flower","flow","flight"};
        System.out.println(new LongestCommonPrefix().longestCommonPrefix(input));
    }

    public String longestCommonPrefix(String[] strs) {
        if(strs==null || strs.length==0) return "";
        String res = strs[0];
        for(int i=1;i<strs.length;i++){
            int v =res.indexOf(strs[i]);
            while(strs[i].indexOf(res)!=0){
                res=(res.substring(0,res.length()-1));
            }
        }
        return res.toString();
    }

}
